File: src/main/java/de/pong/websocket/logic/Particle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package de.pong.websocket.logic;

import java.util.Random;

/**
 *
 * @author simon
 */
public class Particle {

    private int x, y, vx, vy;
    private float radius;

    public Particle(Ball b, Player p) {
        Random r = new Random();

        x = b.getX();
        y = b.getY();
        radius = r.nextInt(5);
        if (p.getX() == 0) {
            vx = r.nextInt(12);
        } else {
            vx = r.nextInt(12) * -1;
        }
        int next = r.nextInt(10);
        if (next < 5) {
            vy = next * -1;
        } else {
            vy = next;
        }
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public float getRadius() {
        return radius;
    }

    public int getVx() {
        return vx;
    }

    public int getVy() {
        return vy;
    }

    public void decreaseRadius() {
        if (radius > 0) {
            radius -= 0.25;
        }
    }

    public void move() {
        x += vx;
        y += vy;
    }

}

File: source/sources/gamelogic/Particle.js
//    private int x, y, vx, vy;
//    private float radius;
//
function Particle(ball, player) {
    this.x = ball.getX();
    this.y = ball.getY();
    this.radius = Math.floor((Math.random() * 5) + 1);

    if (player.getX() == 0) {
        this.vx = Math.floor((Math.random() * 12) + 1);
    } else {
        this.vx = Math.floor((Math.random() * 12) + 1) * -1;
    }

    var next = Math.floor((Math.random() * 10) + 1);
    if (next < 5) {
        this.vy = next * -1;
    } else {
        this.vy = next;
    }
}

Particle.prototype.getX = function () {
    return this.x;
}

Particle.prototype.getY = function () {
    return this.y;
}

Particle.prototype.getRadius = function () {
    return this.radius;
}

Particle.prototype.getVX = function () {
    return this.vx;
}

Particle.prototype.getVY = function () {
    return this.vy;
}

Particle.prototype.decreaseRadius = function () {
    if (this.radius > 0) {
        this.radius -= 0.25;
    }
}

Particle.prototype.move = function () {
    this.x += this.vx;
    this.y += this.vy;
}

module.exports = Particle;